content/en/blog/2023/egress-sni/index.md

```diff
@@ -0,0 +1,441 @@
+---
+title: "[WIP] Routing HTTPS/TLS traffic to arbitrary remote destinations"
+description: "A generic approach to set up egress gateways and use them to route traffic to a restricted set of target remote hosts dynamically, supporting wildcard domains."
+publishdate: 2023-12-10
+attribution: "Gergő Huszty (IBM)"
+keywords: [traffic-management,gateway,mesh,mtls,egress,remote]
+---
+
+If you are using Istio to handle application-originated traffic to destinations outside of the mesh, you're probably familiar with the concept of egress gateways.
+Egress gateways can be used to monitor and forward traffic from mesh-internal applications to locations outside of the mesh.
+This is a useful feature if your system is operating in a restricted
+environment and you want to control what can be reached on the public internet from your mesh.
+
+The use-case of configuring an egress gateway to handle arbitrary wildcard domains had been included in the [official Istio docs](https://archive.istio.io/v1.13/docs/tasks/traffic-management/egress/wildcard-egress-hosts/#wildcard-configuration-for-arbitrary-domains) up until version 1.13, but was subsequently removed because the documented solution was not officially supported or recommended and was subject to breakage in future versions of Istio.
+Nevertheless, the old solution was still usable with Istio versions before 1.20. Istio 1.20, however, dropped some Envoy functionality that was required for the approach to work.
+
+This post attempts to describe how we resolved the issue and filled the gap with a similar approach using Istio version-independent components and Envoy features, but without the need for a separate Nginx SNI proxy.
+Our approach allows users of the old solution to seamlessly migrate configurations before their systems face the breaking changes in Istio 1.20.
+
+## Problem to solve
+
+The currently documented egress gateway use-cases rely on the fact that the target of the traffic
+(the hostname) is statically configured in a `VirtualService`, telling Envoy in the egress gateway pod where to TCP proxy
+the matching outbound connections. You can use multiple, and even wildcard, DNS names to match the routing criteria, but you
+are not able to route the traffic to the exact location specified in the application request. For example you can match traffic for targets
+`*.wikipedia.org`, but you then need to forward the traffic to a single final target, e.g., `en.wikipedia.org`. If there is another
+service, e.g., `anyservice.wikipedia.org`, that is not hosted by the same server(s) as `en.wikipedia.org`, traffic to that host will fail because, even though the target hostname in the
```

Comment:
```suggestion
service, e.g., `anyservice.wikipedia.org`, that is not hosted by the same server(s) as `en.wikipedia.org`, the traffic to that host will fail. This is because, even though the target hostname in the
```
It would be better to separate the long sentence into multiple shorter